### PR TITLE
MES-2499: Add capability to resume to debrief

### PR DESCRIPTION
**Pull request by jameshighfield**
## Description and relevant Jira numbers

Previous test status flow was `Started -> Decided -> Completed -> Submitted`. When the test is on `Decided` and the app is on the Journal page, a "Write Up" button is shown which takes the user to the write up pages. The completed state was previously set when the debrief page is shown.

New test status flow is `Started -> Decided -> WriteUp -> Completed -> Submitted`. Added a new state for `WriteUp` so the `Decided` state can be used to represent the status of the journey between the debrief page and the health declaration page. Now resume launches the test at the pass finalisation page if the status is Decided. This should only happen if the app is force closed.

Hidden the back button from pass finalisation page as the test status did not change on backwards navigation.

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [ ] PR link added to JIRA
- [ ] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

**Comment by jameshighfield** [comment]
Awaiting test on physical device.

**Comment by johnmorleybjss** [comment]
Have you confirmed with design that hiding the Back button on pass finalisation is okay?

**Comment by jameshighfield** [comment]
> Have you confirmed with design that hiding the Back button on pass finalisation is okay?

@johnmorleybjss. Agreed with POs this morning. Will let design know.

File: src/pages/debrief/__tests__/debrief.effects.spec.ts [hunk 1]
```diff
@@ -53,6 +60,22 @@ describe('Debrief Effects', () => {
...
+      // Set activity code as failed
+      actions$.next(new testsActions.SetActivityCode('2'));
+      actions$.next(new debriefActions.EndDebrief());
+
+      effects.endDebriefEffect$.subscribe((result) => {
+        if (result instanceof testStatusActions.SetTestStatusWriteUp) {
+          expect(result).toEqual(new testStatusActions.SetTestStatusWriteUp(currentSlotId));
+        }
+        if (result instanceof testsActions.PersistTests) {
+          expect(result).toEqual(new testsActions.PersistTests());
```
Comment (johnmorleybjss) [inline src/pages/debrief/__tests__/debrief.effects.spec.ts]: As discussed, `done()` below these two if cases will not give us coverage of both branches. As we have guarantee of the new action dispatch order in this case, `done()` can be moved into the second if.

File: src/pages/health-declaration/__tests__/health-declaration.effects.spec.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      actions$.next(new healthDeclarationActions.ContinueFromDeclaration());
+
+      effects.endHealthDeclarationEffect$.subscribe((result) => {
+        if (result instanceof testStatusActions.SetTestStatusWriteUp) {
+          expect(result).toEqual(new testStatusActions.SetTestStatusWriteUp(currentSlotId));
+        }
+        if (result instanceof testsActions.PersistTests) {
+          expect(result).toEqual(new testsActions.PersistTests());
+        }
+        done();
```
Comment (johnmorleybjss) [inline src/pages/health-declaration/__tests__/health-declaration.effects.spec.ts]: Same as above.

File: src/pages/debrief/__tests__/debrief.effects.spec.ts (hunk 1 above)
Comment (szabolcs-keresztes) [reply src/pages/debrief/__tests__/debrief.effects.spec.ts]: We could set a counter and track all the assertion and call `done` when we've done all of them?

Comment (ammarhaiderbjss) [reply src/pages/debrief/__tests__/debrief.effects.spec.ts]: I'll go searching again but IIRC testing for conditional emissions from an effect was unsupported.

Comment (ammarhaiderbjss) [reply src/pages/debrief/__tests__/debrief.effects.spec.ts]: We haven't updated anything so I imagine the case is much the same now.

**Comment by ammarhaiderbjss** [comment]
@jameshighfield have we let the BA's know so they can update their documentation?

**Comment by jameshighfield** [comment]
> @jameshighfield have we let the BA's know so they can update their documentation?

@ammarhaiderbjss I have added a comment to the Jira ticket, tagging a BA

**Pull request merged**